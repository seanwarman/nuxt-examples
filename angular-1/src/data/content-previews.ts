import { RoomPlannerImages, SectionImages } from '../assets/images';

export interface ContentPreview {
  label: string;
  primaryImage: string;
  icon: string;
  body: string;
  slug: string;
  firstSubsection?: string;
  learnIcon: string;
  designIcon: string;
}

export const contentPreviews: ContentPreview[] = [
  {
    slug: 'camera-and-audio',
    label: 'Camera & Audio',
    firstSubsection: 'camera',
    primaryImage: SectionImages.cameraAndAudio.contentPreviewHero,
    icon: SectionImages.cameraAndAudio.icons.sectionIcon,
    designIcon: SectionImages.cameraAndAudio.icons.designIcon,
    learnIcon: SectionImages.cameraAndAudio.icons.learnIcon,
    body: `We've tuned and tweaked the hardware. Now, position it correctly to
     unlock the best performance.`,
  },
  {
    slug: 'displays-and-extras',
    label: 'Displays & Extras',
    firstSubsection: 'screen',
    primaryImage: SectionImages.displaysAndExtras.contentPreviewHero,
    icon: SectionImages.displaysAndExtras.icons.sectionIcon,
    designIcon: SectionImages.displaysAndExtras.icons.designIcon,
    learnIcon: SectionImages.displaysAndExtras.icons.learnIcon,
    body: `Correctly sizing displays is a key part of the conference experience.
      Plus, get some hints and tips on using the Series One Board 65 and Desk 27,
      developed in collaboration with Avocor,
      in your space.`,
  },
  {
    slug: 'room-layout',
    label: 'Room layout',
    firstSubsection: 'viewing-area',
    primaryImage: SectionImages.roomLayout.contentPreviewHero,
    icon: SectionImages.roomLayout.icons.sectionIcon,
    designIcon: SectionImages.roomLayout.icons.designIcon,
    learnIcon: SectionImages.roomLayout.icons.learnIcon,
    body: `Following a few spatial guidelines can make a huge difference for remote and in-room participants.`,
  },
  {
    slug: 'furniture-lighting',
    label: 'Furniture and lighting',
    firstSubsection: 'lighting',
    primaryImage: SectionImages.furnitureLighting.contentPreviewHero,
    icon: SectionImages.furnitureLighting.icons.sectionIcon,
    designIcon: SectionImages.furnitureLighting.icons.designIcon,
    learnIcon: SectionImages.furnitureLighting.icons.learnIcon,
    body: `Make your meeting room look and feel inviting with simple decorative and lighting tips. Appropriate lighting helps both local and remote participants have a great experience.`,
  },
  {
    slug: 'acoustics',
    label: 'Acoustics',
    firstSubsection: 'separation',
    primaryImage: SectionImages.acoustics.contentPreviewHero,
    icon: SectionImages.acoustics.icons.sectionIcon,
    designIcon: SectionImages.acoustics.icons.designIcon,
    learnIcon: SectionImages.acoustics.icons.learnIcon,
    body: `Poor acoustics is a common downfall of video conferencing spaces. The right allowance of acoustic finishes can help make speech clear and audible, and adequate isolation helps avoid interruptions.`,
  },
  {
    slug: 'services',
    label: 'Services',
    firstSubsection: 'air-handling',
    primaryImage: SectionImages.services.contentPreviewHero,
    icon: SectionImages.services.icons.sectionIcon,
    designIcon: SectionImages.services.icons.designIcon,
    learnIcon: SectionImages.services.icons.learnIcon,
    body: `Make sure power is provisioned for the right places, and ensure ventilation systems don't cause unwanted noise pickup on your microphones.`,
  },
  {
    slug: 'room-planner',
    label: 'Room Designer',
    primaryImage: RoomPlannerImages.contentPreviewHero,
    icon: RoomPlannerImages.icons.mainIcon,
    designIcon: SectionImages.services.icons.designIcon,
    learnIcon: SectionImages.services.icons.learnIcon,
    body: `Need to size up your room? All the best practices in this guide are used to help you pick the equipment and recommended room size for the occupancy you need.`,
  },
];
