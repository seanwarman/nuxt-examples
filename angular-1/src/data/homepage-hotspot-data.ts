export const HotspotData: Array<{
  name: string;
  text: string;
  color: string;
  defaultIcon: string;
  interimIcon: string;
  activeIcon: string;
  halo: string;
  x: number;
  y: number;
  z: number;
  cameraPosition: {
    x: number;
    y: number;
    z: number;
  };
  cameraRotation: {
    x: number;
    y: number;
    z: number;
  };
  modelsOn: Array<string>;
  modelsOff: Array<string>;
}> = [
  {
    name: 'camera-and-audio',
    text: 'Camera & Audio',
    color: '#e94233',
    defaultIcon: 'icon-hotspot-camera-audio-red-01-default@3x.png',
    interimIcon: 'icon-hotspot-camera-audio-red-01-interim@3x.png',
    activeIcon: 'icon-hotspot-camera-audio-red-01-active@3x.png',
    halo: 'icon-hotspot-camera-audio-red-01-active-halo@3x.png',
    x: -5,
    y: 113,
    z: -160,
    cameraPosition: {
      x: -100,
      y: 175,
      z: 350,
    },
    cameraRotation: {
      x: 100,
      y: 100,
      z: 0,
    },
    modelsOn: ['CameraAndAudio', 'defaultChairs', 'acousticDefault'],
    modelsOff: ['defaultCameraAndAudio'],
  },
  {
    name: 'displays-and-extras',
    text: 'Displays & Extras',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-displays-extras-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: -230,
    y: 150,
    z: 50,
    cameraPosition: {
      x: 100,
      y: 200,
      z: 500,
    },
    cameraRotation: {
      x: 50,
      y: 100,
      z: 0,
    },
    modelsOn: [
      'DisplaysAndExtras',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'room-layout',
    text: 'Room Layout',
    color: '#31a750',
    defaultIcon: 'icon-hotspot-room-layout-green-01-default@3x.png',
    interimIcon: 'icon-hotspot-room-layout-green-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-layout-green-01-active@3x.png',
    halo: 'icon-hotspot-room-layout-green-01-active-halo@3x.png',
    x: 90,
    y: 20,
    z: 200,
    cameraPosition: {
      x: 0,
      y: 280,
      z: 800,
    },
    cameraRotation: {
      x: 150,
      y: 100,
      z: 100,
    },
    modelsOn: [
      'RoomLayout',
      'defaultChairs',
      'defaultCameraAndAudioSetup',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'furniture-lighting',
    text: 'Furniture & Lighting',
    color: '#fbbc05',
    defaultIcon: 'icon-hotspot-furniture-lighting-yellow-01-default@3x.png',
    interimIcon: 'icon-hotspot-furniture-lighting-yellow-01-interim@3x.png',
    activeIcon: 'icon-hotspot-furniture-lighting-yellow-01-active@3x.png',
    halo: 'icon-hotspot-furniture-lighting-yellow-01-active-halo@3x.png',
    x: 110,
    y: 70,
    z: 80,
    cameraPosition: {
      x: 400,
      y: 200,
      z: 150,
    },
    cameraRotation: {
      x: 0,
      y: 70,
      z: -100,
    },
    modelsOn: [
      'FurnitureAndLighting',
      'defaultCameraAndAudioSetup',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: ['defaultChairs'],
  },
  {
    name: 'acoustics',
    text: 'Acoustics',
    color: '#3d85f2',
    defaultIcon: 'icon-hotspot-acoustics-blue-01-default@3x.png',
    interimIcon: 'icon-hotspot-acoustics-blue-01-interim@3x.png',
    activeIcon: 'icon-hotspot-acoustics-blue-01-active@3x.png',
    halo: 'icon-hotspot-acoustics-blue-01-active-halo@3x.png',
    x: -230,
    y: 200,
    z: 180,
    cameraPosition: {
      x: 300,
      y: 800,
      z: -150,
    },
    cameraRotation: {
      x: -150,
      y: 0,
      z: -100,
    },
    modelsOn: [
      'Acoustics',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticPanel00',
      'acousticPanel01',
      'acousticPanel02',
      'acousticPanel03',
      'defaultCameraAndAudio',
    ],
    modelsOff: ['stripLight00', 'stripLight01', 'acousticDefault'],
  },
  {
    name: 'services',
    text: 'Services',
    color: '#681da7',
    defaultIcon: 'icon-hotspot-services-purple-01-default@3x.png',
    interimIcon: 'icon-hotspot-services-purple-01-interim@3x.png',
    activeIcon: 'icon-hotspot-services-purple-01-active@3x.png',
    halo: 'icon-hotspot-services-purple-01-active-halo@3x.png',
    x: 110,
    y: 115,
    z: -160,
    cameraPosition: {
      x: 300,
      y: 160,
      z: 150,
    },
    cameraRotation: {
      x: 0,
      y: 50,
      z: -250,
    },
    modelsOn: [
      'services',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'room-planner',
    text: 'Room Designer',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-designer-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: -10,
    y: 90,
    z: 0,
    cameraPosition: {
      x: -30,
      y: 150,
      z: 150,
    },
    cameraRotation: {
      x: 40,
      y: 60,
      z: 0,
    },
    modelsOn: [
      'Drawing',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
];
