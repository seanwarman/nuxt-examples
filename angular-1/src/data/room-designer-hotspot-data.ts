export const HotspotData: Array<{
  name: string;
  text: string;
  color: string;
  defaultIcon: string;
  interimIcon: string;
  activeIcon: string;
  halo: string;
  x: number;
  y: number;
  z: number;
  camX: number;
  camY: number;
  camZ: number;
}> = [
  {
    name: 'room-layout',
    text: 'Room Layout',
    color: '#31a750',
    defaultIcon: 'icon-hotspot-room-layout-green-01-default@3x.png',
    interimIcon: 'icon-hotspot-room-layout-green-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-layout-green-01-active@3x.png',
    halo: 'icon-hotspot-room-layout-green-01-active-halo@3x.png',
    x: -110,
    y: 20,
    z: 100,
    camX: -800,
    camY: 800,
    camZ: 800,
  },
  {
    name: 'camera-and-audio',
    text: 'Camera & Audio',
    color: '#e94233',
    defaultIcon: 'icon-hotspot-camera-audio-red-01-default@3x.png',
    interimIcon: 'icon-hotspot-camera-audio-red-01-interim@3x.png',
    activeIcon: 'icon-hotspot-camera-audio-red-01-active@3x.png',
    halo: 'icon-hotspot-camera-audio-red-01-active-halo@3x.png',
    x: -10,
    y: 90,
    z: 0,
    camX: -800,
    camY: 250,
    camZ: 300,
  },
  {
    name: 'displays-and-extras',
    text: 'Displays & Extras',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-displays-extras-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: 0,
    y: 110,
    z: -160,
    camX: -1200,
    camY: 500,
    camZ: 600,
  },
  {
    name: 'services',
    text: 'Services',
    color: '#681da7',
    defaultIcon: 'icon-hotspot-services-purple-01-default@3x.png',
    interimIcon: 'icon-hotspot-services-purple-01-interim@3x.png',
    activeIcon: 'icon-hotspot-services-purple-01-active@3x.png',
    halo: 'icon-hotspot-services-purple-01-active-halo@3x.png',
    x: 120,
    y: 200,
    z: -160,
    camX: -120,
    camY: 250,
    camZ: 300,
  },
  {
    name: 'furniture-lighting',
    text: 'Furniture & Lighting',
    color: '#fbbc05',
    defaultIcon: 'icon-hotspot-furniture-lighting-yellow-01-default@3x.png',
    interimIcon: 'icon-hotspot-furniture-lighting-yellow-01-interim@3x.png',
    activeIcon: 'icon-hotspot-furniture-lighting-yellow-01-active@3x.png',
    halo: 'icon-hotspot-furniture-lighting-yellow-01-active-halo@3x.png',
    x: -70,
    y: 265,
    z: 20,
    camX: -120,
    camY: 300,
    camZ: 400,
  },
  {
    name: 'acoustics',
    text: 'Acoustics',
    color: '#3d85f2',
    defaultIcon: 'icon-hotspot-acoustics-blue-01-default@3x.png',
    interimIcon: 'icon-hotspot-acoustics-blue-01-interim@3x.png',
    activeIcon: 'icon-hotspot-acoustics-blue-01-active@3x.png',
    halo: 'icon-hotspot-acoustics-blue-01-active-halo@3x.png',
    x: -210,
    y: 190,
    z: -160,
    camX: -120,
    camY: 220,
    camZ: 580,
  },
  // {
  //   name: 'room-planner',
  //   text: 'Room Planner',
  //   color: '#0d9cab',
  //   defaultIcon: 'icon-hotspot-teal-01@3x.png',
  //   activeIcon: 'icon-room-designer-white-teal-circle-01@3x.png',
  //   halo: 'teal.png',
  //   x: -10,
  //   y: 150,
  //   z: -160,
  //   camX: 0,
  //   camY: 200,
  //   camZ: 200,
  // },
];
