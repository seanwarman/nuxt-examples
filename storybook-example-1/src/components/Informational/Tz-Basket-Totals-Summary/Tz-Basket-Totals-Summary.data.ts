import { DiscountType } from '@/components/Informational/types';

export const data = {
  noOfItems: 3,
  subTotal: 5100,
  discounts: [
    {
      name: 'C2W discount',
      amount: 1000,
      type: DiscountType.cycleToWork
    },
    {
      name: 'signup10',
      amount: 24.99,
      type: DiscountType.vochure
    },
    {
      name: 'freepump',
      amount: 19.99,
      type: DiscountType.freeGift
    }
  ],
  savings: 1044.98,
  shippingCountries: [
    {
      id: 1,
      name: 'UK mainland'
    },
    {
      id: 2,
      name: 'Northen island'
    },
    {
      id: 3,
      name: 'Scottish highlands'
    },
    {
      id: 4,
      name: 'Scottish islands'
    },
    {
      id: 5,
      name: 'Isle of man'
    }
  ],
  selectedShippingCountry: 1,
  shippingMessage: 'Est, delivery: 15/06/22',
  shippingCost: 0,
  total: 4075.02
};
