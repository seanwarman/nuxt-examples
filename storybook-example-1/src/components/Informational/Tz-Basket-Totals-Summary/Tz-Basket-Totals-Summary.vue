<template>
  <div class="flex flex-col items-stretch gap-6 text-dark">
    <dl class="flex flex-col items-stretch">
      <div class="flex items-center justify-between">
        <dt class="text-sm font-extrabold uppercase">
          Sub total ({{ noOfItems }} {{ noOfItems === 1 ? 'item' : 'items' }})
        </dt>
        <dd class="text-right font-extrabold">{{ handlePrices(subTotal) }}</dd>
      </div>
    </dl>
    <dl class="flex flex-col items-stretch" v-if="discounts">
      <div
        class="flex items-center justify-between"
        v-for="(discount, key) in discounts"
        :key="key"
        :class="{
          'text-primary': discount.type === DiscountType.cycleToWork,
          'text-green-600': discount.type !== DiscountType.cycleToWork
        }"
      >
        <dt class="bg text-sm uppercase">{{ discount.name }}</dt>
        <dd class="text-right">- {{ handlePrices(discount.amount) }}</dd>
      </div>
    </dl>
    <dl class="flex flex-col items-stretch" v-if="savings">
      <div class="flex items-center justify-between">
        <dt class="text-sm font-extrabold uppercase">You save</dt>
        <dd class="text-right font-extrabold">{{ handlePrices(savings) }}</dd>
      </div>
    </dl>
    <section>
      <dl class="flex flex-col items-stretch">
        <div class="flex items-center justify-between">
          <dt class="text-sm font-extrabold uppercase">Delivery</dt>
          <dd class="text-right font-extrabold uppercase">
            {{ shippingCost === 0 ? 'Free' : handlePrices(shippingCost) }}
          </dd>
        </div>
      </dl>
      <div>
        <span class="text-sm text-gray-400">
          <label for="shipping-country">Delivery to </label>
          <select
            id="shipping-country"
            name="shipping-country"
            class="cursor-pointer border-0 p-0 pr-7 text-sm underline"
            v-model="input"
          >
            <option v-for="country in shippingCountries" :key="country.id" :value="country.id">
              {{ country.name }}
            </option>
          </select>
        </span>
        <br />
        <span v-if="shippingMessage" class="text-sm text-gray-400">{{ shippingMessage }}</span>
      </div>
    </section>
    <dl class="flex flex-col items-stretch">
      <div class="flex items-center justify-between">
        <dt class="text-lg font-extrabold uppercase">Total</dt>
        <dd class="text-lg font-extrabold uppercase">{{ handlePrices(total) }}</dd>
      </div>
    </dl>
  </div>
</template>

<script setup lang="ts">
import { handlePrices } from '@/composables/numbers';
import type { DeliverToOption, Discount } from '@/components/Informational/types';
import { DiscountType  } from '@/components/Informational/types';
import { computed } from 'vue';

export type Props = {
  noOfItems: number;
  subTotal: number;
  discounts?: Discount[];
  savings?: number;
  shippingCountries: DeliverToOption[];
  selectedShippingCountry: number;
  shippingCost: number;
  shippingMessage?: string;
  total: number;
};

const props = defineProps<Props>();

const emits = defineEmits<{
  (e: 'deliveryOptionChanged', value: DeliverToOption | undefined): void;
  (e: 'update:selectedShippingCountry', value: number): void;
}>();

const input = computed({
  get: () => props.selectedShippingCountry,
  set: (value) => {
    emits('update:selectedShippingCountry', value);

    const selectedOption = props.shippingCountries.find((x) => x.id === value);
    emits('deliveryOptionChanged', selectedOption);
  }
});
</script>
