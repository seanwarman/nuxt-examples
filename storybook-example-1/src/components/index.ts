export { default as TzButton } from './Action/Tz-Button/Tz-Button.vue';
export { default as TzCheckoutCta } from './Action/Tz-Checkout-CTA/Tz-Checkout-Cta.vue';
export { default as TzModalDialog } from './Action/Tz-ModalDialog/Tz-ModalDialog.vue';
export { default as TzPayPalButton } from './Action/Tz-Button/Tz-PayPalButton.vue';
export { default as TzProgressBar } from './Action/Tz-Progress-Bar/Tz-Progress-Bar.vue';

export { default as TzBasketAccordion } from './Informational/Tz-Basket-Accordion/Tz-Basket-Accordion.vue';
export { default as TzBasketCard } from './Informational/Tz-Basket-Card/Tz-Basket-Card.vue';
export { default as TzBasketTotalsSummary } from './Informational/Tz-Basket-Totals-Summary/Tz-Basket-Totals-Summary.vue';
export { default as TzBundleCreator } from './Informational/Tz-BundleCreator/Tz-BundleCreator.vue';
export { default as TzC2wBanner } from './Informational/Tz-C2W-Banner/Tz-C2W-Banner.vue';
export { default as TzExpressCheckout } from './Informational/Tz-ExpressCheckout/Tz-ExpressCheckout.vue';
export { default as TzInfoBox } from './Informational/Tz-InfoBox/Tz-InfoBox.vue';
export { default as TzJumboCard } from './Informational/Tz-Jumbo-Card/Tz-Jumbo-Card.vue';
export { default as TzOrderItemsSummary } from './Informational/Tz-Order-Items-Summary/Tz-Order-Items-Summary.vue';
export { default as TzOrderTotalsSummary } from './Informational/Tz-Order-Totals-Summary/Tz-Order-Totals-Summary.vue';
export { default as TzProductTile } from './Informational/Tz-Product-Tile/Tz-Product-Tile.vue';
export { default as TzUspRibbon } from './Informational/Tz-USP-Ribbon/Tz-USP-Ribbon.vue';

export { default as TzInputField } from './Input/Tz-InputField/Tz-InputField.vue';
export { default as TzLogin } from './Input/Form/Tz-Login/Tz-Login.vue';
export { default as TzPaymentMethods } from './Input/Tz-PaymentMethods/Tz-PaymentMethods.vue';
export { default as TzRadioField } from './Input/Tz-RadioField/Tz-RadioField.vue';
export { default as TzSelectField } from './Input/Tz-SelectField/Tz-SelectField.vue';

export { default as TzNavigation } from './Navigation/Tz-Navigation/Tz-Navigation.vue';

export { default as TzContainer } from './Structural/Tz-Container/Tz-Container.vue';

export { default as TzAccountIcon } from './Informational/Icons/Tz-Account-Icon.vue';
export { default as TzArrowIcon } from './Informational/Icons/Tz-Arrow-Icon.vue';
export { default as TzBasketIcon } from './Informational/Icons/Tz-Basket-Icon.vue';
export { default as TzCaretIcon } from './Informational/Icons/Tz-Caret-Icon.vue';
export { default as TzCheckmarkIcon } from './Informational/Icons/Tz-Checkmark-Icon.vue';
export { default as TzChevronIcon } from './Informational/Icons/Tz-Chevron-Icon.vue';
export { default as TzCloseIcon } from './Informational/Icons/Tz-Close-Icon.vue';
export { default as TzExclaimationIcon } from './Informational/Icons/Tz-Exclaimation-Icon.vue';
export { default as TzExpandIcon } from './Informational/Icons/Tz-Expand-Icon.vue';
export { default as TzFilterIcon } from './Informational/Icons/Tz-Filter-Icon.vue';
export { default as TzSearchIcon } from './Informational/Icons/Tz-Search-Icon.vue';
export { default as TzWishlistIcon } from './Informational/Icons/Tz-Wishlist-Icon.vue';
