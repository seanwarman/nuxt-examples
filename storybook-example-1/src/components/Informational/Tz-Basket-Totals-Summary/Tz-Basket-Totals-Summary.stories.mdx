import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';
import { action } from '@storybook/addon-actions';

import { DiscountType } from '@/components/Informational/types';
import TzBasketTotalsSummary from './Tz-Basket-Totals-Summary.vue';
import { data } from './Tz-Basket-Totals-Summary.data.ts';
import { MobileViewports } from '@/helpers/viewports';

<Meta title="components/Informational/Basket Totals Summary" component={TzBasketTotalsSummary} />

export const Template = (args) => ({
  components: { TzBasketTotalsSummary },
  setup() {
    return { args };
  },
  template: `<tz-basket-totals-summary v-bind="args" @deliveryOptionChanged="deliveryOptionChanged">{{ args.text }}</tz-basket-totals-summary>`,
  methods: {
    deliveryOptionChanged: action('deliveryOptionChanged')
  }
});

# Basket Totals Summary

Intended to be used in the basket where shipping country can be updated.

When the user selects a new delivery location, the `deliveryOptionChanged` event fires and will need to trigger a recalulation
of the total values in the parent view. This component is "dumb" and doesn't perform any calculations itself.

<Canvas withSource>
  <Story name="Default" args={data}>
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Default" />

## With selected delivery option

With a selected delivery option set, the delivery option is fixed.

<Canvas withSource>
  <Story
    name="With selected delivery option"
    args={{
      ...data,
      ...{
        selectedShippingCountry: 3
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## With no discounts

<Canvas withSource>
  <Story
    name="With no discounts"
    args={{
      ...data,
      ...{
        discounts: null,
        savings: 0,
        total: 5100
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>
