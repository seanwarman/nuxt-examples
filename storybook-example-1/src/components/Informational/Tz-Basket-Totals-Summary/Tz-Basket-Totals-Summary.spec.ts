/**
 * @vitest-environment jsdom
 */

import { VueWrapper, mount } from '@vue/test-utils';
import { expect, describe, beforeEach, it } from 'vitest';
import { TzBasketTotalsSummary } from '@/components';

import { data as props } from './Tz-Basket-Totals-Summary.data';

let wrapper: VueWrapper;

describe('TzInfoBox', () => {
  beforeEach(() => {
    wrapper = mount(TzBasketTotalsSummary, {
      props
    });
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("emits 'deliveryOptionChanged' event when shipping country is changed", async () => {
    wrapper.find('#shipping-country').setValue(3);
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('deliveryOptionChanged')).toBeTruthy();
    expect(wrapper.emitted('deliveryOptionChanged')![0]).toEqual([props.shippingCountries[2]]);
  });
});
