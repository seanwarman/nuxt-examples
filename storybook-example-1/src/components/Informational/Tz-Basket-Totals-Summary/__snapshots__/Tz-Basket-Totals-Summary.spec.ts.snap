// Vitest Snapshot v1

exports[`TzInfoBox > renders correctly 1`] = `
<div
  class="flex flex-col items-stretch gap-6 text-dark"
>
  <dl
    class="flex flex-col items-stretch"
  >
    <div
      class="flex items-center justify-between"
    >
      <dt
        class="text-sm font-extrabold uppercase"
      >
         Sub total (3 items) 
      </dt>
      <dd
        class="text-right font-extrabold"
      >
        £5,100.00
      </dd>
    </div>
  </dl>
  <dl
    class="flex flex-col items-stretch"
  >
    
    <div
      class="flex items-center justify-between text-primary"
    >
      <dt
        class="bg text-sm uppercase"
      >
        C2W discount
      </dt>
      <dd
        class="text-right"
      >
        - £1,000.00
      </dd>
    </div>
    <div
      class="flex items-center justify-between text-green-600"
    >
      <dt
        class="bg text-sm uppercase"
      >
        signup10
      </dt>
      <dd
        class="text-right"
      >
        - £24.99
      </dd>
    </div>
    <div
      class="flex items-center justify-between text-green-600"
    >
      <dt
        class="bg text-sm uppercase"
      >
        freepump
      </dt>
      <dd
        class="text-right"
      >
        - £19.99
      </dd>
    </div>
    
  </dl>
  <dl
    class="flex flex-col items-stretch"
  >
    <div
      class="flex items-center justify-between"
    >
      <dt
        class="text-sm font-extrabold uppercase"
      >
        You save
      </dt>
      <dd
        class="text-right font-extrabold"
      >
        £1,044.98
      </dd>
    </div>
  </dl>
  <section>
    <dl
      class="flex flex-col items-stretch"
    >
      <div
        class="flex items-center justify-between"
      >
        <dt
          class="text-sm font-extrabold uppercase"
        >
          Delivery
        </dt>
        <dd
          class="text-right font-extrabold uppercase"
        >
          Free
        </dd>
      </div>
    </dl>
    <div>
      <span
        class="text-sm text-gray-400"
      >
        <label
          for="shipping-country"
        >
          Delivery to 
        </label>
        <select
          class="cursor-pointer border-0 p-0 pr-7 text-sm underline"
          id="shipping-country"
          name="shipping-country"
        >
          
          <option
            value="1"
          >
            UK mainland
          </option>
          <option
            value="2"
          >
            Northen island
          </option>
          <option
            value="3"
          >
            Scottish highlands
          </option>
          <option
            value="4"
          >
            Scottish islands
          </option>
          <option
            value="5"
          >
            Isle of man
          </option>
          
        </select>
      </span>
      <br />
      <span
        class="text-sm text-gray-400"
      >
        Est, delivery: 15/06/22
      </span>
    </div>
  </section>
  <dl
    class="flex flex-col items-stretch"
  >
    <div
      class="flex items-center justify-between"
    >
      <dt
        class="text-lg font-extrabold uppercase"
      >
        Total
      </dt>
      <dd
        class="text-lg font-extrabold uppercase"
      >
        £4,075.02
      </dd>
    </div>
  </dl>
</div>
`;
